app/controllers/publishers/ats_api/v1/create_vacancy_service.rb

```diff
@@ -0,0 +1,77 @@
+module Publishers
+  module AtsApi
+    module V1
+      class CreateVacancyService
+        InvalidOrganisationError = Class.new(StandardError)
+
+        class << self
+          def call(params)
+            vacancy = Vacancy.new(permitted_params(params))
+
+            if (conflict = conflict_vacancy(vacancy))
+              return conflict_response(conflict)
+            end
+
+            if vacancy.save
+              success_response(vacancy)
+            else
+              validation_error_response(vacancy)
+            end
+          end
+
+          private
+
+          def permitted_params(params)
+            organisations = fetch_organisations(params[:schools])
+            raise InvalidOrganisationError, "No valid organisations found" if organisations.blank?
+
+            params[:publish_on] ||= Time.zone.today.to_s
+            params[:working_patterns] ||= []
+            params[:phases] ||= []
+
+            params.except(:schools, :trust_uid).merge(organisations: organisations)
+          end
+
+          def fetch_organisations(school_params)
```

Comment:
could we remove the duplication of this method between the create and the update?

Reply:
IMHO this is suggesting that we have 1 service with 2 entry points (create and update). My challenge with it is that is had already diverged, potentially creating strange 'bugs' e.g.404 on update with invalid org, 400 on create with invalid org.